src/filesystem/index.ts

```diff
@@ -577,6 +716,349 @@ server.setRequestHandler(CallToolRequestSchema, async (request) => {
+                  type: "text",
+                  text: `Loaded file ${uriPath}`
+                }
+              });
+            }
+          }
+        }
+
+        // Start recursive processing from the root directory
+        await processDirectory(validPath);
+
+        if (messages.length === 0) {
+          throw new Error("No readable files found in directory");
+        }
+
+        messages.push({
+          role: "assistant",
+          content: {
+            type: "text",
+            text: `Loaded ${messages.length / 2} files recursively from ${validPath}`
+          }
+        });
+        return { messages };
+      }
+
+      default:
+        throw new Error(`Unknown prompt name: ${name}`);
+    }
+  } catch (error) {
+    const errorMessage = error instanceof Error ? error.message : String(error);
+    throw new Error(`Failed to handle prompt ${request.params.name}: ${errorMessage}`);
+  }
+});
+
+server.setRequestHandler(ListPromptsRequestSchema, async () => {
+  return {
+    prompts: [
+      {
+        name: 'load_file',
+        description: "Load and analyze contents of a single file with automatic type detection and appropriate handling for text, image, and binary files",
+        arguments: [
+          {
+            name: "path",
+            description: "Path to the file relative to workspace root",
+            type: "string",
+            required: true,
+          }
+        ]
+      },
+      {
+        name: 'load_folder',
+        description: "Recursively load and analyze all files in a directory and its subdirectories (limited to 30 files maximum). Automatically detects and appropriately handles different file types, skipping unsupported ones.",
+        arguments: [
+          {
+            name: "path",
+            description: "Path to the directory relative to workspace root",
+            type: "string",
+            required: true,
+          }
+        ]
+      }
+    ]
+  };
+});
+
+// Define the FileType enum
+enum FileType {
+  TEXT = 'TEXT',
+  IMAGE = 'IMAGE',
+  BINARY = 'BINARY',
+}
+
+// Utility function to determine file type
+function loadAsType(mimeType: string, stats: Stats | BigIntStats): FileType {
+  if (mimeType.startsWith('text/') || 
+      ['application/json', 'application/javascript', 'application/typescript', 'application/xml', 'application/csv', 'application/tsv'].includes(mimeType)) {
+    return FileType.TEXT;
+  }
+
+  if (mimeType.startsWith('image/')) {
+    return FileType.IMAGE;
+  }
+
+  if (['application/octet-stream', 'application/pdf', 'application/zip', 'application/x-tar', 'application/x-gzip', 'application/x-bzip2', 'application/x-7z-compressed', 'application/x-rar', 'application/x-xz'].includes(mimeType)) {
+    return FileType.BINARY;
+  }
+
+  // If file is smaller than 100kb and type is unknown, treat as text
+  if (stats.size < 100 * 1024) {
+    return FileType.TEXT;
+  }
+
+  // Otherwise fall back to binary
+  return FileType.BINARY;
```

Comment:
TIOLI: Best strategy here is probably to try reading it as a string, and if that fails, fall back to binary. This is something we could do later, though.

Reply:
I _think_ it will always be possible to "read" any file "as a string", just the string might be absolute garbage and weird characters.

I was thinking about some options around this
 - have separate read_binary, read_image, read_text functions - the caller will generally know what they want and could keep this approach as a `read_auto` or someting

- alternatively make an optional input argument for binary/image/text and fall back to this behaviour if it's not specified